src/connector/src/source/kafka/source/reader.rs

```diff
@@ -150,6 +150,10 @@ impl SplitReader for KafkaSplitReader {
 
         consumer.assign(&tpl)?;
 
+        // poll consumer to trigger callback functions
+        // We don't need to wait for the result, because recv may block and wait for next new message
+        let _ = tokio::time::timeout(Duration::from_millis(1000), consumer.recv()).await;
+
```

Comment:
Will this read a new message and ignore it?

Reply:
after adjusting the code position, the `recv` call will not dismiss the first message.

Reply:
https://github.com/risingwavelabs/risingwave/pull/21868/commits/4d50e8d055ee0b916d8c9c7d76e6e1bafc5db315 from https://github.com/risingwavelabs/risingwave/pull/21868 also do the same change. already contained in image [v2.2.6-patch-kafka-poll-log-level-3](https://github.com/orgs/risingwavelabs/packages/container/risingwave/417466703?tag=v2.2.6-patch-kafka-poll-log-level-3)